docs/common_hal.md

```diff
@@ -0,0 +1,70 @@
+# Adding `nativeio` support to other ports
```

Comment:
@willingc Thank you for the awesome suggestions! I incorporated them all except for the title change. I didn't change it because I don't want this "how to port" guide confused with the `nativeio` reference docs with a similar title.